### PR TITLE
Doc update 003

### PR DESCRIPTION
**Pull request by ChrisBradley001**
This doc update includes various corrections or additions resulting from a technical review of the doc set.

### CODE REVIEW

**Comment by terracotta-org** [comment]
Can one of the admins verify this patch?

File: docs/src/docs/asciidoc/user/getting-started.adoc [hunk 1]
```diff
@@ -21,7 +21,7 @@ NOTE: If you are looking to use the JSR-107, aka `javax.cache` API, you should s
 [[configuring-with-java]]
 === Configuring with Java
 
-Java configuration is most easily achieved through the use of builders that offer a fluent API.
+Java configuration is most easily achieved through the use of builders that offer a straightforward API.
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: fluent is the technical term for the kind of interface we use (`build.withThis().thanThat()`). So I think we want to keep fluent here

File: docs/src/docs/asciidoc/user/getting-started.adoc [hunk 2]
```diff
@@ -32,23 +32,28 @@ As with the previous versions of Ehcache, the canonical way of dealing with `Cac
...
-<3> Finally, invoking `.build()` returns a fully instantiated, but uninitialized, `CacheManager` we can use;
-<4> Before you start to use the `CacheManager` it needs to be `init()`, which can be done for you by the builder by passing
-    `true` to `build(boolean)`;
-<5> We can retrieve the `preConfigured` aliased `Cache` we declared in step 2. For type-safety, we ask for both key and
-    value types to be passed in. If these differ from the ones we expect, the `CacheManager` throws a `ClassCastException`
-    early in the application's lifecycle. It also guards the `Cache` from being polluted by random types.
+<1> The static method `org.ehcache.config.builders.CacheManagerBuilder.newCacheManagerBuilder` returns a new `org.ehcache.config.builders.CacheManagerBuilder` instance.
+<2> Use the builder to define a `Cache` with alias "preConfigured". This cache will be created when `cacheManager.build()` is invoked on the actual `CacheManager` instance. 
+The first String argument is the cache alias, which is used to retrieve the cache from the `CacheManager`.
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: `String` (with quote)

File: docs/src/docs/asciidoc/user/getting-started.adoc [hunk 3]
```diff
@@ -32,23 +32,28 @@ As with the previous versions of Ehcache, the canonical way of dealing with `Cac
...
-    `true` to `build(boolean)`;
-<5> We can retrieve the `preConfigured` aliased `Cache` we declared in step 2. For type-safety, we ask for both key and
-    value types to be passed in. If these differ from the ones we expect, the `CacheManager` throws a `ClassCastException`
-    early in the application's lifecycle. It also guards the `Cache` from being polluted by random types.
+<1> The static method `org.ehcache.config.builders.CacheManagerBuilder.newCacheManagerBuilder` returns a new `org.ehcache.config.builders.CacheManagerBuilder` instance.
+<2> Use the builder to define a `Cache` with alias "preConfigured". This cache will be created when `cacheManager.build()` is invoked on the actual `CacheManager` instance. 
+The first String argument is the cache alias, which is used to retrieve the cache from the `CacheManager`. 
+The second argument, `org.ehcache.config.CacheConfiguration`, is used to configure the `Cache`. 
+We use the static `.newCacheConfigurationBuilder()` method on `org.ehcache.config.builders.CacheConfigurationBuilder` to create a default configuration.
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: Remove the . before newCacheConfigurationBuilder

File: docs/src/docs/asciidoc/user/getting-started.adoc [hunk 4]
```diff
@@ -32,23 +32,28 @@ As with the previous versions of Ehcache, the canonical way of dealing with `Cac
...
+<3> Finally, invoking `.build()` returns a fully instantiated, but uninitialized, `CacheManager` we can use.
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: Same. Remove the dot before build()

File: docs/src/docs/asciidoc/user/getting-started.adoc [hunk 5]
```diff
@@ -32,23 +32,28 @@ As with the previous versions of Ehcache, the canonical way of dealing with `Cac
...
+<4> Before using the `CacheManager` it needs to be initialized, which can be done in 1 of 2 ways: 
+Calling `CacheManager.init()` on the `CacheManager` instance, or calling the `CacheManagerBuilder.build(boolean init)` method with the boolean parameter set to true.
+<5> A cache is retrieved by passing its alias, key type and value type to the `CacheManager`. 
+For instance, to obtain the cache declared in step 2 you need its alias=preConfigured, keyType=Long.class and valueType=String.class. 
+For type-safety, we ask for both key and value types to be passed in. 
+If these differ from the ones we expect, the `CacheManager` throws a `ClassCastException` early in the application's lifecycle. 
+It also guards the `Cache` from being polluted by random types.
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: Too much `also` (here and below)

File: docs/src/docs/asciidoc/user/getting-started.adoc [hunk 6]
```diff
@@ -32,23 +32,28 @@ As with the previous versions of Ehcache, the canonical way of dealing with `Cac
...
 <6> The `CacheManager` can also be used to create new `Cache` instances as needed. Just as in step 2, it requires passing in an
     alias as well as a `CacheConfiguration`. The instantiated and fully initialized `Cache` added will be returned and/or
     accessed through the `CacheManager.getCache` API.
-<7> We can now use the newly added `Cache` to store and ...
-<8> ... retrieve data.
+<7> The newly added `Cache` can now be used to store entries, which are comprised of key value pairs. 
+The put method's first parameter is the key and the second parameter is the value.
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: I'm not and English native but I thought `'s` was only for people

File: docs/src/docs/asciidoc/user/getting-started.adoc [hunk 7]
```diff
@@ -125,7 +130,10 @@ include::{sourcedir32}/impl/src/test/java/org/ehcache/docs/GettingStarted.java[t
 
 <1> If you wish to use disk storage (like for persistent `Cache` instances), you'll have to provide a
     location where data should be stored on disk to the `CacheManagerBuilder.persistence(String)` static method.
-<2> You define a resource pool for the disk.
+<2> Defines a resource pool for the disk. 
+The third parameter is a boolean value which is used to set whether the disk pool is persistent. 
+When set to true the pool is persistent, and when set to false the pool is not persistent.
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: Comma after "true"

File: docs/src/docs/asciidoc/user/xml.adoc [hunk 8]
```diff
@@ -16,15 +16,14 @@ Using an XML file you can configure a `CacheManager` at creation time, according
 
 === `<config>` root element
 
-The root element of our XML configuration. One `<config>` element and, by implication, one XML file,
-provides the definition for a `CacheManager`.  With Ehcache 3, however, you may create multiple
-`CacheManager` instances using the same XML configuration file.  Unlike the JSR-107
-`javax.cache.spi.CachingProvider`, Ehcache does not maintain a registry of `CacheManager` instances.
+The root element of our XML configuration. One `<config>` element in an XML file provides the definition for a `CacheManager`.  
 
+Note: Ehcache allows for creating multiple `CacheManager` instances using the same XML configuration file.
```
Comment (henri-tremblay) [inline docs/src/docs/asciidoc/user/xml.adoc]: Maybe put Note in bold

Comment (ljacomet) [inline docs/src/docs/asciidoc/user/xml.adoc]: I would instead use NOTE in all caps, which generates a note box, as used further down in this document.

File: docs/src/docs/asciidoc/user/serializers-copiers.adoc [hunk 9]
```diff
@@ -9,7 +9,7 @@ ifdef::notBuildingForSite[]
 include::menu.adoc[]
 endif::notBuildingForSite[]
 
-While Ehcache is a Java cache, it cannot always store its mapping as Java objects.
+While Ehcache is a Java cache, it cannot always store its mappings as Java objects.
```
Comment (ljacomet) [inline docs/src/docs/asciidoc/user/serializers-copiers.adoc]: This change also exists in #1644, that's going to cause useless conflicts.

File: docs/src/docs/asciidoc/user/getting-started.adoc [hunk 10]
```diff
@@ -32,23 +32,28 @@ As with the previous versions of Ehcache, the canonical way of dealing with `Cac
...
-    argument is `org.ehcache.config.CacheConfiguration` to configure the `Cache`. We use the static
-    `.newCacheConfigurationBuilder()` method on `org.ehcache.config.builders.CacheConfigurationBuilder` to create a default config;
-<3> Finally, invoking `.build()` returns a fully instantiated, but uninitialized, `CacheManager` we can use;
-<4> Before you start to use the `CacheManager` it needs to be `init()`, which can be done for you by the builder by passing
-    `true` to `build(boolean)`;
-<5> We can retrieve the `preConfigured` aliased `Cache` we declared in step 2. For type-safety, we ask for both key and
-    value types to be passed in. If these differ from the ones we expect, the `CacheManager` throws a `ClassCastException`
-    early in the application's lifecycle. It also guards the `Cache` from being polluted by random types.
+<1> The static method `org.ehcache.config.builders.CacheManagerBuilder.newCacheManagerBuilder` returns a new `org.ehcache.config.builders.CacheManagerBuilder` instance.
+<2> Use the builder to define a `Cache` with alias "preConfigured". This cache will be created when `cacheManager.build()` is invoked on the actual `CacheManager` instance.
```
Comment (albinsuresh) [inline docs/src/docs/asciidoc/user/getting-started.adoc]: Isn't it wrong to say that "the cache will be created when `cacheManager.build()` is invoked"? The actual cache doesn't get created on `build()` call. Only the configuration gets built. The actual cache is created in the `cacheManager.init()` call.

**Comment by ljacomet** [comment]
Replaced with #1704 and #1705

**Pull request closed without merging**